dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/AdrServiceClient.cs

```diff
@@ -31,197 +34,292 @@ public async ValueTask DisposeAsync()
         }
 
         await _assetServiceClient.DisposeAsync().ConfigureAwait(false);
-        await _assetEndpointProfileServiceClient.DisposeAsync().ConfigureAwait(false);
+        await _deviceServiceClient.DisposeAsync().ConfigureAwait(false);
         GC.SuppressFinalize(this);
         _disposed = true;
     }
 
-    /// <inheritdoc/>
-    public async Task<NotificationResponse> ObserveAssetEndpointProfileUpdatesAsync(string aepName, TimeSpan? commandTimeout = null,
+    /// <inheritdoc />
+    public async Task<NotificationResponse> ObserveDeviceEndpointUpdatesAsync(string deviceName, string endpointName, TimeSpan? commandTimeout = null,
         CancellationToken cancellationToken = default)
     {
         cancellationToken.ThrowIfCancellationRequested();
         ObjectDisposedException.ThrowIf(_disposed, this);
 
-        _observedAeps[aepName] = _dummyByte;
-        await _assetServiceClient.AssetEndpointProfileUpdateEventTelemetryReceiver.StartAsync(cancellationToken);
+        _observedEndpoints[$"{deviceName}_{endpointName}"] = _dummyByte;
```

Comment:
The service-side docs refer to this field as "InboundEndpointName" rather than just "EndpointName" so we should probably follow suit